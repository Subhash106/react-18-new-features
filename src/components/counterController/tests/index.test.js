import renderer from "react-test-renderer";
import Counter from "../counter";

it("It should render the snapshot", () => {
  const component = renderer.create(<Counter person="Subash" />);
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();

  console.log(tree.props);

  // trigger action
  renderer.act(() => {
    tree.props.onMouseEnter();
  });
  //re-render
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // trigger action
  renderer.act(() => {
    tree.props.onMouseLeave();
  });
  //re-render
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
