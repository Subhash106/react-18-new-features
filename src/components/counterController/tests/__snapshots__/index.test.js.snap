// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render the snapshot 1`] = `
<div
  className="counter"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <p>
    Subash's count
  </p>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    Add 1
  </button>
</div>
`;

exports[`It should render the snapshot 2`] = `
<div
  className="counter hover"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <p>
    Subash's count
  </p>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    Add 1
  </button>
</div>
`;

exports[`It should render the snapshot 3`] = `
<div
  className="counter"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <p>
    Subash's count
  </p>
  <p>
    0
  </p>
  <button
    onClick={[Function]}
  >
    Add 1
  </button>
</div>
`;
